File: game-server/src/main/java/work/fking/pangya/game/packet/outbound/PangBalancePacket.java
package work.fking.pangya.game.packet.outbound;

import io.netty.buffer.ByteBuf;
import work.fking.pangya.game.player.Player;
import work.fking.pangya.networking.protocol.OutboundPacket;

public class PangBalancePacket implements OutboundPacket {

    private static final int ID = 0xc8;

    private final int balance;

    public PangBalancePacket(int balance) {
        this.balance = balance;
    }

    public static PangBalancePacket create(Player player) {
        return new PangBalancePacket(player.pangBalance());
    }

    @Override
    public void encode(ByteBuf target) {
        target.writeShortLE(ID);
        target.writeLongLE(balance);
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/packet/outbound/PangBalancePacket.kt
package work.fking.pangya.game.packet.outbound

import io.netty.buffer.ByteBuf
import work.fking.pangya.game.player.Player
import work.fking.pangya.networking.protocol.OutboundPacket

class PangBalancePacket(
    private val balance: Int
) : OutboundPacket {

    constructor(player: Player) : this(player.pangBalance())

    override fun encode(target: ByteBuf) {
        target.writeShortLE(0xc8)
        target.writeLongLE(balance.toLong())
    }
}
